service/resource/s3objectv1/spec.go

```diff
@@ -2,7 +2,15 @@ package s3objectv1
 
 import "github.com/aws/aws-sdk-go/service/s3"
 
-type BucketObject struct {
+const (
+	prefixWorker = "worker"
+)
+
+type BucketObjectState struct {
+	WorkerCloudConfig BucketObjectInstance
+}
```

Comment:
We could also have an array of bucket objects? Fine with this for now and we can revisit when we add the master cloud config.

Reply:
Maybe a `map[string]BucketObjectInstance`? IMO with an array we would depend on order, but introducing a map would lead us to implement checks for existent keys, WDYT? We can discuss later in any case.

Reply:
Yes, a map is better and prevents ordering problems.